content/2017/03/21/blocking-analytics-spam.rst

```diff
@@ -0,0 +1,145 @@
+public: yes
+author: jonny
+tags: [
+  Spam,
+  DevOps,
+  'Build Tools',
+  'Front-end',
+  JavaScript,
+  Analytics,
+  Code
+  ]
+image:
+  - src: '2017/server-client/spa2.jpg'
+summary: |
+  Google Analytics is great for gathering data on who uses your web
+  application, but becomes worthless if spam sessions start infesting your
+  data. Here's how we've tried to combat the problem for `oddbird.net`_.
+
+  .. _oddbird.net: /
+
+
+Blocking Analytics Spam
+=======================
+
+Like many websites, we use `Google Analytics`_ to gather data about our users –
+what OS and browser they used, how they came to our site, etc. But a number of
+months ago we started seeing lots of this:
+
+.. _Google Analytics: https://analytics.google.com/
+
+.. image:: /static/images/blog/2017/ga-spam/ga-spam.jpg
+   :alt: google analytics spam
+   :class: img-border
+
+It's not a new problem, but it's particularly problematic for smaller sites
+that don't receive lots of traffic. On a given day, spam hits were accounting
+for anywhere from 10–90(!) percent of our sessions.
```

Comment:
Stylistically, I prefer this written out: "from ten to ninety (!) percent". But you may not!